lib/compat/wordpress-trunk/widget-render-api-endpoint.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/**
+ * Polyfill API class to render widgets via the REST API.
+ *
+ * @see \WP_REST_Controller
+ */
+class WP_REST_Widget_Render_Endpoint_Polyfill extends \WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wp/v2';
+		$this->rest_base = 'widget-types';
+		$this->full_endpoint_instance = new WP_REST_Widget_Types_Controller();
```

Comment:
This is my way of reusing `get_item_permissions_check`. It should be safe because `WP_REST_Widget_Types_Controller` is either supplied by core or the Gutenberg plugin.

Reply:
I thought about extending `WP_REST_Widget_Types_Controller` but I don't like bringing over all that code just to be able to check permissions. This implementation here is more isolated.